sale_promotion_rule/models/sale_order.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    promotion_rule_id = fields.Many2one(
+        'sale.promotion.rule',
+        string='Promotion rule')
+
+    @api.multi
+    def clear_promotion_line(self):
+        for order in self:
+            for line in order.order_line:
```

Comment:
I would do it in a more compact way:
`order.order_line.filtered('promotion_rule_id')`